ql/ql/src/codeql_ql/ast/internal/Builtins.qll

```diff
@@ -94,3 +94,148 @@ class FloatClass extends PrimitiveType {
+      or
+      i = 2 and
+      result instanceof EquivalenceRelation::EquivalenceRelationModule
+    }
+  }
+
+  /**
+   * A mock that implements the `EquivalenceRelation` module.
+   * The equivalent to the following is implemented:
+   * ```CodeQL
+   * module QlBuiltins {
+   *  signature class T;
+   *  module EdgeSig<T MyT> { // This might not be needed.
+   *    signature predicate edgeSig(MyT a, MyT b);
+   *  }
+   *  module EquivalenceRelation<T MyT, EdgeSig<MyT>::edgeSig/2 edge> { // the `edge` parameter is not modeled
+   *    class EquivalenceClass;
+   *    EquivalenceClass getEquivalenceClass(MyT a);
+   *  }
+   *}
+   */
+  module EquivalenceRelation {
+    class SigClass extends MockClass::Range {
+      SigClass() { this = "Mock: QlBuiltins::T" }
+
+      override string getName() { result = "T" }
+    }
+
+    /** A mock TypeExpr with classname `T`. */
+    class DummyTTypeExpr extends MockTypeExpr::Range {
+      DummyTTypeExpr() { this = "Mock: QlBuiltins::T" }
+
+      override string getClassName() { result = "T" }
+    }
+
+    module EdgeSig {
+      class EdgeSigModule extends MockModule::Range {
+        EdgeSigModule() { this = "Mock: QlBuiltins::EdgeSig" }
+
+        override string getName() { result = "EdgeSig" }
+
+        override predicate hasTypeParam(int i, string type, string name) {
+          i = 0 and name = "MyT" and type instanceof DummyTTypeExpr
+        }
+
+        override string getMember(int i) { i = 0 and result instanceof EdgeSigPred }
+      }
+
+      class EdgeSigPred extends MockClasslessPredicate::Range {
+        EdgeSigPred() { this = "Mock: QlBuiltins::EdgeSig::edgeSig" }
+
+        override string getName() { result = "edgeSig" }
+
+        override EdgeSigPredParam getParameter(int i) {
+          i = 0 and
+          result.getName() = "a"
+          or
+          i = 1 and
+          result.getName() = "b"
+        }
+      }
+
+      class EdgeSigPredParam extends MockVarDecl::Range {
+        string name;
+
+        EdgeSigPredParam() {
+          this = "Mock: QlBuiltins::EdgeSig::edgeSig::" + name and name = ["a", "b"]
+        }
+
+        override string getName() { result = name }
+
+        override MockTypeExpr::Range getType() { result instanceof DummyTTypeExpr }
+      }
+    }
+
+    class EquivalenceRelationModule extends MockModule::Range {
+      EquivalenceRelationModule() { this = "Mock: QlBuiltins::EquivalenceRelation" }
+
+      override string getName() { result = "EquivalenceRelation" }
+
+      override string getMember(int i) {
+        i = 0 and result instanceof EquivalenceClassClass
+        or
+        i = 1 and result instanceof GetEquivalenceClassPredicate
+      }
+
+      override predicate hasTypeParam(int i, string type, string name) {
+        i = 0 and name = "MyT" and type instanceof DummyTTypeExpr
+        or
+        none() // TODO: `EdgeSig<MyT>::edgeSig/2 edge` is not implemented.
```

Comment:
should we implement this, or would you like to post-pone that?

Reply:
I don't think it's even needed.  

I left the comment there because that type-parameter is not implemented.   
But that only becomes an issue if QL-for-QL decides to check if the number of type-parameters match the type-arguments at the instantiation, which it currently doesn't.  
So I want to keep the comment, in case we starting doing such a check, but we don't need to do anything about it now.